### PR TITLE
Support Silex 2

### PR DESCRIPTION
**Pull request by sergiors**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -13,3 +13,4 @@ build/
 vendor/
 composer.phar
 composer.lock
+composer.lock
```
Comment (ttskch) [inline .gitignore]: no need to add

File: composer.json [hunk 2]
```diff
@@ -14,12 +14,12 @@
         }
     ],
     "require": {
-        "php": ">=5.3.3",
-        "silex/silex": "~1.0",
+        "php": ">=5.5",
```
Comment (ttskch) [inline composer.json]: [`>=5.5.9` is correct](https://github.com/silexphp/Silex/blob/master/composer.json#L18) I think

File: .gitignore (hunk 1 above)
Comment (sergiors) [inline .gitignore]: done

File: composer.json (hunk 2 above)
Comment (sergiors) [inline composer.json]: done!

File: src/Provider/PaginationServiceProvider.php [hunk 3]
```diff
@@ -8,102 +9,103 @@
...
         if (!isset($app['twig'])) {
-            $app->register(new TwigServiceProvider());
-        }
-        if (!isset($app['url_generator'])) {
-            $app->register(new UrlGeneratorServiceProvider());
+            throw new \LogicException('You must register the TwigServiceProvider to use the PaginationServiceProvider.');
         }
         if (!isset($app['translator'])) {
-            $app->register(new TranslationServiceProvider());
+            throw new \LogicException('You must register the TranslationServiceProvider to use the PaginationServiceProvider.');
```
Comment (ttskch) [inline src/Provider/PaginationServiceProvider.php]: `TwigServiceProvider` and `TranslationServiceProvider` are included in Silex2.0, so there is no problem to automatically register them here I think. How about that?

Comment (ttskch) [inline src/Provider/PaginationServiceProvider.php]: also [`RoutingServiceProvider`](https://github.com/silexphp/Silex/blob/master/src/Silex/Provider/RoutingServiceProvider.php#L48) for `$app['url_generator']`

**Comment by ttskch** [comment]
@sergiors Thanks a lot for your PR! I commented about something. Please review them 🙏

File: src/Provider/PaginationServiceProvider.php (hunk 3 above)
Comment (sergiors) [inline src/Provider/PaginationServiceProvider.php]: It's not a problem. But I Think `PaginationServiceProvider` doesn't have responsibility to register `TwigServiceProvider` and  `TranslationServiceProvider`, those provires are dependencies, and should be defined at userland.

Like silex providers
https://github.com/silexphp/Silex/blob/master/src/Silex/Provider/TranslationServiceProvider.php#L34

Comment (ttskch) [inline src/Provider/PaginationServiceProvider.php]: Thanks, I understand! 😃

Comment (sergiors) [inline src/Provider/PaginationServiceProvider.php]: I think `$app['url_generator']` belongs to `RoutingServiceProvider`, make sense, register them.
`TwigServiceProvider ` and `TranslationServiceProvider` doesn't belongs to `PaginationServiceProvider`, but him needs the dependencies

Comment (ttskch) [inline src/Provider/PaginationServiceProvider.php]: You're right 👍  And I found `RoutingServiceProvider` is [automatically registered in C'tor of `Silex\Application`](https://github.com/silexphp/Silex/blob/master/src/Silex/Application.php#L70) so `PaginationServiceProvider` doesn't need register it expressly I think.

**Comment by ttskch** [comment]
All clear! I'm merging it. Thanks a lot 🌟

**Pull request merged**